Infrastructure/InMemoryUserRepository.cs

```diff
@@ -0,0 +1,123 @@
+﻿using Application;
+using Domain;
+using Domain.Exceptions;
+
+namespace Infrastructure
+{
+    public class InMemoryUserRepository : IUserRepository
+    {
+        private readonly List<User> _users = new();
+
+        public void AddUsers(List<User> users)
+        {
+            foreach (User user in users)
+            {
+                _users.Add(user);
+            }
+        }
+
+        public void AddUser(User user)
+        {
+            _users.Add(user);
+        }
+
+        public void RemoveUser(User user)
+        {
+            _users.Remove(user);
+        }
+
+        public List<string> GetAllDisplayNames()
+        {
+            List<string> displayNames = new();
+
+            foreach (User user in _users)
+            {
+                displayNames.Add(user.DisplayName);
+            }
+
+            return displayNames;
+        }
+
+        public List<int> GetAllIds()
+        {
+            List<int> ids = new();
+
+            foreach (User user in _users)
+            {
+                ids.Add(user.Id);
+            }
+
+            return ids;
+        }
+
+        public User GetUserByUsernameAndPassword(string username, string password)
+        {
+            User loggedUser = _users.Find((user) => username == user.Username && password == user.Password);
+
+            return loggedUser == null ? throw new UserNotFoundException(username) : loggedUser;
```

Comment:
You can also return null and check for null after submitting login info to avoid throwing an exception.

Reply:
Updated code.
Check "renamed AcceptOrRemoveFriendRequest to UpdateFriendRequest, renamed..." commit